libraries/core_libs/network/src/tarcap/packets_handlers/pbft_sync_packet_handler.cpp

```diff
@@ -67,6 +67,10 @@ void PbftSyncPacketHandler::process(const PacketData &packet_data, const std::sh
     }
   }
 
+  if (!pbft_mgr_->checkBlockWeight(sync_block)) {
+    throw MaliciousPeerException("Block summary gas amount is bigger then limit");
+  }
+
```

Comment:
Estimations wouldn't change. Because we are estimating it with proposal period from dag block, not on the latest block state(see [block_proposer.cpp:209](https://github.com/Taraxa-project/taraxa-node/pull/1672/files#diff-0240e8dd854ad98bf711fcd683da658fe26250cc6bf29d2c54b488b8a96dba60R209))